### DISCUSSION TITLE
Automated End-to-end IBC tests

### DISCUSSION THREAD

**Discussion by michaelfig** [Agoric/agoric-sdk#5479] [Ideas]
End-to-end IBC tests.

We would like to work together with Strangelove to use their [Go-orchestrated Docker test framework](https://github.com/strangelove-ventures/ibc-test-framework) to test:

- Agoric-to-Cosmos outbound channel is automatically relayed
  - Agoric-to-Cosmos mangled packet errors are propagated back to the sender
- Agoric-to-Cosmos bidirectional fungible token transfer via Pegasus
- Agoric-to-Cosmos Interchain accounts
- Hermes relayer works in addition to Go relayer

This epic will result in more work items.

**Comment by dckc** [comment]
we're cheering on Jack Z. and gang

**Comment by dckc** [comment]
seems to have come together in [multichain-testing](https://github.com/Agoric/agoric-sdk/tree/master/multichain-testing)
